hub_sdk/modules/projects.py

```diff
@@ -49,12 +49,32 @@ def get_data(self) -> None:
         Returns:
             (None): The method does not return a value.
         """
-        if self.id:
-            resp = super().read(self.id).json()
-            self.data = resp.get("data", {})
-            self.logger.debug("Project id is %s", self.id)
-        else:
+        if not self.id:
             self.logger.error("No project id has been set. Update the project id or create a project.")
+            return
+
+        try:
+            response = super().read(self.id)
+
+            if response is None:
+                self.logger.error("Received no response from the server for project id %s", self.id)
+                return
+
+            # Check if the response has a .json() method (it should if it's a response object)
+            if not hasattr(response, "json"):
+                self.logger.error("Invalid response object received for project id %s", self.id)
+                return
+
+            resp_data = response.json()
+            if resp_data is None:
+                self.logger.error("No data received in the response for project id %s", self.id)
+                return
+
+            self.data = resp_data.get("data", {})
+            self.logger.debug("Project data retrieved for id %s", self.id)
+
+        except Exception as e:
+            self.logger.error("An error occurred while retrieving data for project id %s: %s", self.id, str(e))
```

Comment:
@hassaanfarooq01 we should use fstrings in all places now, the percentage sign usage is legacy and should not be used for new code, i.e.:

```diff
self.logger.error("An error occurred while retrieving data for project id %s: %s", self.id, str(e))
self.logger.error(f"An error occurred while retrieving data for project id {self.id}: {e}")
```

Can you apply fstrings to the 10 logger lines in the PR? Then it should be good to go. Thanks!

Reply:
@glenn-jocher Yeah Sure